### PR TITLE
Fix and polish artifact dockerfiles (and make them buildable again)

### PR DESCRIPTION
**Pull request by jtattermusch**
Currently,  `apt-get install -y crossbuild-essential-armhf` doesn't work on debian jessie because the packages are broken (see https://lists.debian.org/debian-embedded/2018/03/msg00000.html),
which prevents are from rebuilding the artifact docker image.

Steps taken to make the dockerfile buildable again:
- move Node and its dependencies to a separate docker image, which builds on top of debian:stretch (Node will likely need to update its artifact building scripts.
- other languages (C#, PHP, ruby) continue using the jessie-based image (it's hard to switch all of them to debian stretch, I tried).
- while doing this, perform a bit of cleanup as well.

This is needed for some followup work on C# artifact building.

### CODE REVIEW

**Comment by jtattermusch** [comment]
@nicolasnoble  @mehrdada   please give owners approval.

File: tools/dockerfile/grpc_artifact_node_linux_x64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Copyright 2016 gRPC authors.
```
Comment (mehrdada) [inline tools/dockerfile/grpc_artifact_node_linux_x64/Dockerfile]: s/2016 gRPC authors./2018 The gRPC Authors/

(same for the other file)

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by murgatroid99** [comment]
I'm pretty sure we're not actually using those Dockerfiles at all for Node, so those new ones shouldn't be necessary. @nicolasnoble you put together the new Node build process, can you confirm?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by jtattermusch** [comment]
@murgatroid99  ok, I removed the node docker files.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,802      Total (=)      1,964,802

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,887,470      Total (>)     10,887,469

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,964,802      Total (=)      1,964,802

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,887,471      Total (>)     10,887,469

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by nicolasnoble** [comment]
The artifact build has migrated to the grpc-node repository, alongside its Dockerfiles.

**Pull request merged**